domains/transfer-agreements/TransferAgreements.API/API.UnitTests/Services/ProjectOriginWalletServiceTest.cs

```diff
@@ -0,0 +1,133 @@
+    {
+        var transferAgreement = new TransferAgreement()
+        {
+            Id = Guid.NewGuid(), StartDate = DateTimeOffset.UtcNow, SenderName = "Producent A/S",
+            SenderTin = "32132112", ReceiverTin = "11223344"
+        };
+
+        var fakeGranularCertificatesResponse = CreateAsyncUnaryCall(
+            new QueryResponse { GranularCertificates = { CreateGranularCertificate(DateTimeOffset.UtcNow.AddDays(1), DateTimeOffset.UtcNow.AddDays(2))  } }
+        );
+
+
+        var fakeTransferResponse = CreateAsyncUnaryCall(
+            new TransferResponse() {}
+        );
+
+        SetupWalletServiceClient(fakeGranularCertificatesResponse, fakeTransferResponse);
+
+        await service.TransferCertificates(transferAgreement);
+
+        fakeWalletServiceClient.Received(1).TransferCertificateAsync(Arg.Any<TransferRequest>(), Arg.Any<Metadata>());
+    }
+
+    [Fact]
+    public async Task TransferCertificates_CertificateStartIsInvalidAndTANoEndDate_ShouldNotCallWalletTransferCertificate()
+    {
+        var transferAgreement = new TransferAgreement()
+        {
+            Id = Guid.NewGuid(), StartDate = DateTimeOffset.UtcNow, SenderName = "Producent A/S",
+            SenderTin = "32132112", ReceiverTin = "11223344"
+        };
+
+        var fakeGranularCertificatesResponse = CreateAsyncUnaryCall(
+            new QueryResponse { GranularCertificates = { CreateGranularCertificate(DateTimeOffset.UtcNow.AddDays(-2), DateTimeOffset.UtcNow.AddDays(-1))  } }
+        );
+
+        var fakeTransferResponse = CreateAsyncUnaryCall(
+            new TransferResponse() {}
+        );
+
+        SetupWalletServiceClient(fakeGranularCertificatesResponse, fakeTransferResponse);
+
+        await service.TransferCertificates(transferAgreement);
+
+        fakeWalletServiceClient.Received(0).TransferCertificateAsync(Arg.Any<TransferRequest>(), Arg.Any<Metadata>());
+    }
+
+    [Fact]
+    public async Task TransferCertificates_WhenCalled_ShouldCallWalletTransferCertificate()
+    {
+        var transferAgreement = new TransferAgreement()
+        {
+            Id = Guid.NewGuid(), StartDate = DateTimeOffset.UtcNow, EndDate = DateTimeOffset.UtcNow.AddDays(3), SenderName = "Producent A/S",
+            SenderTin = "32132112", ReceiverTin = "11223344"
+        };
+
+        var fakeGranularCertificatesResponse = CreateAsyncUnaryCall(
+            new QueryResponse { GranularCertificates = { CreateGranularCertificate(DateTimeOffset.UtcNow.AddDays(1), DateTimeOffset.UtcNow.AddDays(2))  } }
+        );
+
+        var fakeTransferResponse = CreateAsyncUnaryCall(
+            new TransferResponse() {}
+        );
+        SetupWalletServiceClient(fakeGranularCertificatesResponse, fakeTransferResponse);
+
+
+        await service.TransferCertificates(transferAgreement);
+
+        fakeWalletServiceClient.Received(1).TransferCertificateAsync(Arg.Any<TransferRequest>(), Arg.Any<Metadata>());
+    }
```

Comment:
Maybe a test for if certificate enddate is after TA start date, but the certificate startdate is not

Reply:
fixed in c68f1480